src/resolver/findAllComponentDefinitions.js

```diff
@@ -34,7 +35,19 @@ export default function findAllReactCreateClassCalls(
     return false;
   }
 
+  function statelessVisitor(path) {
+    if (isStatelessComponent(path)) {
+      // TODO: normalizeStatelessDefinition to pick up propTypes
+      definitions.push(path);
+    }
+    return false;
+  }
+
   recast.visit(ast, {
+    visitFunctionDeclaration: statelessVisitor,
+    visitFunctionExpression: statelessVisitor,
+    visitProperty: statelessVisitor,
```

Comment:
I think this should be handled by the other visitors since methods are just represented as function expressions in the AST.

On Oct 7, 2015, at 12:51 PM, Dustan Kasten <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/resolver/findAllComponentDefinitions.js<https://github.com/reactjs/react-docgen/pull/28#discussion_r41437429>:

>    recast.visit(ast, {
> +    visitFunctionDeclaration: statelessVisitor,
> +    visitFunctionExpression: statelessVisitor,
> +    visitProperty: statelessVisitor,


I added this as part of supporting some of the permutations also being added to gaearon/babel-plugin-react-transform; namely this format: https://github.com/gaearon/babel-plugin-react-transform/pull/34/files#diff-9d6f6ff3fa6b5b63c09190bd7a2efb95R15

Honestly I’m not sure if that is captured or handled correctly yet. [:innocent:]

—
Reply to this email directly or view it on GitHub<https://github.com/reactjs/react-docgen/pull/28/files#r41437429>.